exercises/binary-search-tree/Example.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public class BinarySearchTree<T>
+    where T : IComparable
```

Comment:
The exercise only talks about using numbers so there's no point in making this generic. I would keep it simple and just make this non-generic and use `int`s.